File: Presentations/PEPM2020/projectile2/python/Calculations.py
import math

def func_t_flight(v_launch, theta, g_vect):
    outfile = open("log.txt", "a")
    print("function func_t_flight called with inputs: {", file=outfile)
    print("  v_launch = ", end='', file=outfile)
    print(v_launch, end='', file=outfile)
    print(", ", file=outfile)
    print("  theta = ", end='', file=outfile)
    print(theta, end='', file=outfile)
    print(", ", file=outfile)
    print("  g_vect = ", end='', file=outfile)
    print(g_vect, file=outfile)
    print("  }", file=outfile)
    outfile.close()
    
    return 2 * v_launch * math.sin(theta) / g_vect

def func_p_land(v_launch, theta, g_vect):
    outfile = open("log.txt", "a")
    print("function func_p_land called with inputs: {", file=outfile)
    print("  v_launch = ", end='', file=outfile)
    print(v_launch, end='', file=outfile)
    print(", ", file=outfile)
    print("  theta = ", end='', file=outfile)
    print(theta, end='', file=outfile)
    print(", ", file=outfile)
    print("  g_vect = ", end='', file=outfile)
    print(g_vect, file=outfile)
    print("  }", file=outfile)
    outfile.close()
    
    return 2 * v_launch ** 2 * math.sin(theta) * math.cos(theta) / g_vect

def func_d_offset(p_target, p_land):
    outfile = open("log.txt", "a")
    print("function func_d_offset called with inputs: {", file=outfile)
    print("  p_target = ", end='', file=outfile)
    print(p_target, end='', file=outfile)
    print(", ", file=outfile)
    print("  p_land = ", end='', file=outfile)
    print(p_land, file=outfile)
    print("  }", file=outfile)
    outfile.close()
    
    return p_land - p_target

def func_s(p_target, epsilon, d_offset):
    outfile = open("log.txt", "a")
    print("function func_s called with inputs: {", file=outfile)
    print("  p_target = ", end='', file=outfile)
    print(p_target, end='', file=outfile)
    print(", ", file=outfile)
    print("  epsilon = ", end='', file=outfile)
    print(epsilon, end='', file=outfile)
    print(", ", file=outfile)
    print("  d_offset = ", end='', file=outfile)
    print(d_offset, file=outfile)
    print("  }", file=outfile)
    outfile.close()
    
    if (math.fabs(d_offset / p_target) < epsilon) :
        return "The target was hit."
    elif (d_offset < 0) :
        return "The projectile fell short."
    else :
        return "The projectile went long."

File: Presentations/PEPM2020/projectile2/cpp/Calculations.cpp
#include "Calculations.hpp"

#include <fstream>
#include <iostream>
#include <math.h>
#include <string>

using std::ofstream;
using std::string;

double func_t_flight(double v_launch, double theta, double g_vect) {
    ofstream outfile;
    outfile.open("log.txt", std::fstream::app);
    outfile << "function func_t_flight called with inputs: {" << std::endl;
    outfile << "  v_launch = ";
    outfile << v_launch;
    outfile << ", " << std::endl;
    outfile << "  theta = ";
    outfile << theta;
    outfile << ", " << std::endl;
    outfile << "  g_vect = ";
    outfile << g_vect << std::endl;
    outfile << "  }" << std::endl;
    outfile.close();
    
    return 2 * v_launch * sin(theta) / g_vect;
}

double func_p_land(double v_launch, double theta, double g_vect) {
    ofstream outfile;
    outfile.open("log.txt", std::fstream::app);
    outfile << "function func_p_land called with inputs: {" << std::endl;
    outfile << "  v_launch = ";
    outfile << v_launch;
    outfile << ", " << std::endl;
    outfile << "  theta = ";
    outfile << theta;
    outfile << ", " << std::endl;
    outfile << "  g_vect = ";
    outfile << g_vect << std::endl;
    outfile << "  }" << std::endl;
    outfile.close();
    
    return 2 * pow(v_launch, 2) * sin(theta) * cos(theta) / g_vect;
}

double func_d_offset(double p_target, double p_land) {
    ofstream outfile;
    outfile.open("log.txt", std::fstream::app);
    outfile << "function func_d_offset called with inputs: {" << std::endl;
    outfile << "  p_target = ";
    outfile << p_target;
    outfile << ", " << std::endl;
    outfile << "  p_land = ";
    outfile << p_land << std::endl;
    outfile << "  }" << std::endl;
    outfile.close();
    
    return p_land - p_target;
}

string func_s(double p_target, double epsilon, double d_offset) {
    ofstream outfile;
    outfile.open("log.txt", std::fstream::app);
    outfile << "function func_s called with inputs: {" << std::endl;
    outfile << "  p_target = ";
    outfile << p_target;
    outfile << ", " << std::endl;
    outfile << "  epsilon = ";
    outfile << epsilon;
    outfile << ", " << std::endl;
    outfile << "  d_offset = ";
    outfile << d_offset << std::endl;
    outfile << "  }" << std::endl;
    outfile.close();
    
    if (fabs(d_offset / p_target) < epsilon) {
        return "The target was hit.";
    }
    else if (d_offset < 0) {
        return "The projectile fell short.";
    }
    else {
        return "The projectile went long.";
    }
}
